src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -93,6 +94,37 @@ public static bool RunningOnAlpine
             }
         }
 
+        public static bool RunningOnAppleSiliconAsX64
+        {
+            get
+            {
+                if (RunningOnMacOS)
+                {
+                    try
+                    {
+                        // See https://stackoverflow.com/questions/65259300/detect-apple-silicon-from-command-line
+                        ProcessStartInfo psi = new ProcessStartInfo
+                        {
+                            FileName = "/bin/sh",
+                            Arguments = "-c \"sysctl -n machdep.cpu.brand_string\"",
+                            RedirectStandardOutput = true,
+                            UseShellExecute = false
+                        };
+
+                        var process = Process.Start(psi);
+                        var cpuBrand = process.StandardOutput.ReadToEnd();
+                        return cpuBrand.Contains("Apple") && RuntimeInformation.ProcessArchitecture != Architecture.Arm;
```

Comment:
can this process block the agent forever?
e.g., chcp can do this: https://github.com/microsoft/azure-pipelines-agent/pull/4760
consider using the ProcessInvoker + cancellation token + retries for each external tool calls

Reply:
I wasn't aware of the ReadToEnd issue 😮 
P.s. We probably need some code best practices guide in the repo 😃